From: Greg Kroah-Hartman <gregkh@suse.de>
Date: Thu, 3 Dec 2009 16:24:47 -0800
Subject: [PATCH 2/4] Staging: samsung-laptop: remove old kernel code

commit 5b4c4dc61d5445346b1d5465c4b4934ea933165b upstream.

Don't test for the kernel version, we know what version we are in,
the latest.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/samsung-laptop/samsung-laptop.c |   40 -----------------------
 1 files changed, 0 insertions(+), 40 deletions(-)

diff --git a/drivers/staging/samsung-laptop/samsung-laptop.c b/drivers/staging/samsung-laptop/samsung-laptop.c
index 253b447..4877138 100644
--- a/drivers/staging/samsung-laptop/samsung-laptop.c
+++ b/drivers/staging/samsung-laptop/samsung-laptop.c
@@ -9,7 +9,6 @@
  * the Free Software Foundation.
  *
  */
-#include <linux/version.h>
 #include <linux/kernel.h>
 #include <linux/init.h>
 #include <linux/module.h>
@@ -274,7 +273,6 @@ static struct backlight_ops backlight_ops = {
 	.update_status	= update_status,
 };
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,31)
 static int rfkill_set(void *data, bool blocked)
 {
 	/* Do something with blocked...*/
@@ -318,44 +316,6 @@ static void destroy_wireless(void)
 	rfkill_destroy(rfk);
 }
 
-#else
-
-static int rfkill_set(void *data, enum rfkill_state state)
-{
-	if (state ==  RFKILL_STATE_UNBLOCKED)
-		sabi_set_command(SET_WIRELESS_BUTTON, 1);
-	else
-		sabi_set_command(SET_WIRELESS_BUTTON, 0);
-
-	return 0;
-}
-
-static int init_wireless(struct platform_device *sdev)
-{
-	int retval;
-
-	rfk = rfkill_allocate(&sdev->dev, RFKILL_TYPE_WLAN);
-	if (!rfk)
-		return -ENOMEM;
-	rfk->toggle_radio = rfkill_set;
-	rfk->name = "samsung-wifi";
-
-	retval = rfkill_register(rfk);
-	if (retval) {
-		rfkill_free(rfk);
-		return -ENODEV;
-	}
-
-	return 0;
-}
-
-static void destroy_wireless(void)
-{
-	rfkill_unregister(rfk);
-}
-
-#endif
-
 static ssize_t get_silent_state(struct device *dev,
 				struct device_attribute *attr, char *buf)
 {
-- 
1.7.1

